File: src/Classes/Encounter/Encounter.ts
export default class Encounter {

    id: number;
    entities: Array<Entity>;
    items: Array<Item>;
    notes: Array<string>;
    cr: number;
    exp: number;
    name: string;
    description: string;
    imageUrl: string;

    constructor(id: number, entites: Array<Entity>, items: Array<Item>, notes: Array<string>,
        cr: number, exp: number, name: string, description: string, imageUrl: string) {
        this.id = id;
        this.entities = entites;
        this.items = items;
        this.notes = notes;
        this.cr = cr;
        this.exp = exp;
        this.name = name;
        this.description = description;
        this.imageUrl = imageUrl;
    }
}

export class Entity {
    id: number;
    name: string;
    description: string;
    apiUrl: string;

    constructor(id: number, name: string, description: string, apiUrl: string) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.apiUrl = apiUrl;
    }
}

export class Item {
    id: number;
    name: string;
    description: string;
    apiUrl: string;

    constructor(id: number, name: string, description: string, apiUrl: string) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.apiUrl = apiUrl;
    }
}

File: server/Models/Encounter.cs
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;
using System.Collections.Generic;

namespace dnd_buddy.Models
{

    public class Encounter
    {
        [Key]
        public int Id { get; set; }
        public List<Entity> Entities { get; set; }
        public List<string> Notes { get; set; }
        public int CR { get; set; }
        public int Exp { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string ImageUrl { get; set; }
        public Encounter()
        {
            this.Entities = new List<Entity>();
            this.Notes = new List<string>();
            this.CR = 0;
            this.Exp = 0;
            this.Name = "New Encounter";
            this.Description = "";
            this.ImageUrl = "";
        }
        public class Entity
        {
            [Key]
            public int Id { get; set; }
            public string Name { get; set; }
            public string Description { get; set; }
            public string ApiUrl { get; set; }
            public Entity()
            {
                this.Name = "New Entity";
                this.Description = "";
                this.ApiUrl = "";
            }
        }
        public class Item
        {
            [Key]
            public int Id { get; set; }
            public string Name { get; set; }
            public string Description { get; set; }
            public string ApiUrl { get; set; }
            public Item()
            {
                this.Name = "New Entity";
                this.Description = "";
                this.ApiUrl = "";
            }
        }
    }
}